src/main/java/killua/parser/Parser.java

```diff
@@ -0,0 +1,242 @@
+     * @return The Event object created from the arguments.
+     * @throws KilluaException If the arguments are empty or in an incorrect format.
+     */
+    public static Event parseEvent(String arguments) throws KilluaException {
+        if (arguments.isEmpty()) {
+            throw new KilluaException("Arguments cannot be empty!");
+        }
+
+        try {
+            String[] parts = arguments.split(" /from ");
+            String description = parts[0].strip();
+            String[] dateStrings = parts[1].split(" /to ");
+
+            String fromDateTimeString = dateStrings[0].strip();
+            String toDateTimeString = dateStrings[1].strip();
+
+            try {
+                LocalDateTime fromDateTime = LocalDateTime.parse(fromDateTimeString, DATE_TIME_FORMATTER);
+                LocalDateTime toDateTime = LocalDateTime.parse(toDateTimeString, DATE_TIME_FORMATTER);
+                return new Event(description, fromDateTime, toDateTime);
+            } catch (DateTimeParseException e) {
+                LocalDate fromDate = LocalDate.parse(fromDateTimeString, DATE_FORMATTER);
+                LocalDate toDate = LocalDate.parse(toDateTimeString, DATE_FORMATTER);
+                return new Event(description, fromDate, toDate);
+            }
+        } catch (ArrayIndexOutOfBoundsException | DateTimeParseException e) {
+            throw new KilluaException("Please use the correct format for events: event <description> /from <yyyy-mm-dd> /to <yyyy-mm-dd> OR event <description> /from <yyyy-mm-dd hh:mm> /to <yyyy-mm-dd hh:mm>");
+        }
+    }
+
+    /**
+     * Parses an index argument and returns the corresponding integer index.
+     *
+     * @param arguments The index argument.
+     * @return The integer index parsed from the arguments.
+     * @throws KilluaException If the arguments are empty or not a valid integer.
+     */
+    public static int parseIndex(String arguments) throws KilluaException {
+        if (arguments.isEmpty()) {
+            throw new KilluaException("Arguments cannot be empty!");
+        }
+
+        try {
+            return Integer.parseInt(arguments.trim()) - 1;
+        } catch (NumberFormatException e) {
+            throw new KilluaException("Please provide a valid task number!");
+        }  catch (IndexOutOfBoundsException e1) {
+            throw new KilluaException("Task " + arguments + " not found!");
+        }
+    }
+
+    /**
+     * Parses a task string from the storage format and returns the corresponding Task object.
+     *
+     * @param line The task string from the storage.
+     * @return The Task object parsed from the string.
+     * @throws IllegalArgumentException If the task type is unknown.
+     */
+    public static Task parseTask(String line) {
+        char taskType = line.charAt(0);
+        boolean isDone = line.charAt(4) == '1';
+        String argument = line.substring(8);
+
+        Task task;
+
+        if (taskType == 'T') {
```

Comment:
Perhaps you can use a switch statement here to avoid the many `if` `else` clauses.